src/Avalonia.Controls.TreeDataGrid/ITreeDataGridSource.cs

```diff
@@ -31,6 +31,10 @@ public interface ITreeDataGridSource
         /// </summary>
         ITreeDataGridSelection? Selection { get; }
 
+        IEnumerable<object> Items { get; }
+
+        IEnumerable<object>? GetModelChildren(object model);
```

Comment:
A bit confusing that it returns a nullable enumerable. When even for the flat source it returns an empty enumerable instead.

Reply:
Probably simple docs explaining return value will be nice to have

Reply:
> A bit confusing that it returns a nullable enumerable. When even for the flat source it returns an empty enumerable instead.

That's to keep compatibility with the existing code in which the existing method is nullable.  See https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/30902a7345a93a58cb3598bfbc4bf71ba9632c25/src/Avalonia.Controls.TreeDataGrid/HierarchicalTreeDataGridSource.cs#L184

I'd change it to be non-nullable, but a cascade of changes would be needed, then.

Reply:
> Probably simple docs explaining return value will be nice to have



> Probably simple docs explaining return value will be nice to have

Do you mean "summary" documentation?

Reply:
/// <summary> ...  </summary>
/// <returns> ... </returns>
IEnumerable<TModel>? GetModelChildren(TModel model);

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#returns

Reply:
Added! :)

Reply:
Yeah this is nullable because it's just exposing an underlying property on the model, and we don't want to enforce that the model returns a non-null child collection because `null` and `[]` may mean different things.